### ISSUE TITLE
[research] AERA-v26.10 stage/control-flow idle attribution after #732/#735

### ISSUE THREAD

**Issue by vinceackermann2-sys** [vinceackermann2-sys/tam-research#736]
## Purpose

Resolve the remaining systems ambiguity after authoritative #732/#735 returned `next_target=None`. The side-by-side CUDA timeline shows that no common named kernel or repeated kernel family meets the preregistered materiality rule, while distortion-normalized inter-device idle dominates both batches (~47.16 ms candidate at batch 8 and ~58.13 ms candidate at batch 64). Before another optimization, localize that idle/host/control time to concrete execution regions.

This issue is diagnostic-only. It initially authorizes ZERO GPU work and ZERO optimization.

## Exact source and parent evidence

Bind live source at issue creation:
- `SOURCE_MAIN=d15d94f39e59d0c6cc8d1514f44a33452d279a74`
- `SOURCE_TREE=fe5181034b3cb07076b17a43c7137829124d8cbb`

Authoritative parent diagnostic:
- issue #732
- sole L4 trigger #735
- evidence comment `5574994790`
- workflow run `34156438962`
- job `101849137998`
- attempt `1`
- device NVIDIA L4
- parent `next_target=None`
- no optimization/E2E systems/higher-stage authority.

Parent candidate observations:
- batch 8 unprofiled median `52.51891326904297 ms`; profiled `89.11689758300781 ms`; `1716` kernels; normalized idle gap `47.15760479756141 ms`.
- batch 64 unprofiled median `101.74742126464844 ms`; profiled `126.5898208618164 ms`; `1755` kernels; normalized idle gap `58.12677116908063 ms`.
- batch 8 largest common named kernel is a single CUTLASS GEMM ~1.54 ms normalized; repeated BF16 copy kernel ~294 calls but only ~0.45 ms.
- batch 64 largest common named kernel is a single CUTLASS GEMM ~19.52 ms; repeated BF16 copy ~303 calls / ~4.82 ms.
- no single common kernel/kernel-cluster met the frozen materiality rule across both batches.

Historical exclusions remain binding:
- #405: launch-count/event reduction alone can be slower; do not infer latency benefit from fewer launches.
- #258: reasoner-only `torch.compile` is insufficient; do not repeat compile-alone.
- #731: latent-depth synchronization coalescing is scientifically FAIL as a >=5% full-call optimization.

## Diagnostic question

On the exact frozen v26.10 candidate used by #731/#735, which execution region is responsible for the largest common material **host/control/device-idle** contribution at both batch 8 and batch 64?

The diagnostic must convert whole-call idle into region-localized evidence without changing learned equations, routing decisions, physical sparsity, checkpoint bytes, parameters, durable state, or model outputs.

## Frozen execution fixtures

Reuse exactly:
- frozen #710 memory-safe launcher blob `b885250753ea169cd89dd7a978bb3647fd261fe8`;
- frozen v26.10 implementation blob `d8f691c198eed1fa96bcbb78a4e76cad82d18779`;
- seed8471 checkpoints/hashes;
- batches `(8,64)`;
- token seed rule inherited from #710/#731;
- `route_mode="hard_sparse"`, `hard=True`, `update_memory=True`;
- CUDA BF16 autocast through the frozen model call;
- no training, optimizer, backward, checkpoint write, corpus access, or new scientific seed.

Before attribution, require the same #731 correctness gates: exact route/expert/depth decisions and active-row populations, schema/weight/checkpoint identity, finite outputs, logits/state equivalence under `atol=rtol=1e-2`, and no dense-masked sparse credit.

## Region boundary

Prefer additive instrumentation only (wrappers/hooks/context managers in the diagnostic harness), not runtime/model source edits. Use explicit nested `torch.profiler.record_function` annotations and preserve original call order/results.

At minimum distinguish these stable families when present:
1. `chunk_outer_glue` — chunk embedding/input-state assembly and between-stage glue not owned by a stage module;
2. `stage_route_control` — optional-stage gate/selection/compaction/scatter control around stage invocation;
3. `stage_compute` — each actual stage invocation, further decomposed into:
   - `norm_start_controller`
   - `context_attention`
   - `expert_dispatch_compute`
   - `end_controller`
   - `latent_reasoner`
   - `reason_to_chunk_output_norm`
   - `recurrent_stream_update`
   - `write_select_update`
4. within `expert_dispatch_compute`, where safely hookable without changing semantics:
   - first-expert selection/gather/compute
   - second-expert route/selection/compute/reduction
5. within `latent_reasoner`, distinguish depth-schedule/control from GRU-cell compute if the frozen v26.10 implementation exposes a safe boundary;
6. `ficem_read_write` where the backend exposes a stable call boundary;
7. `durable_state_pack_output` — final state/output assembly after stage work.

If exact sub-boundaries cannot be observed without modifying runtime semantics, mark them unavailable; do not infer ownership.

## Measurement method

For each frozen batch:
- 3 warmup calls;
- 3 separately measured unprofiled CUDA-event whole-call calls;
- exactly 1 Kineto profiled whole call with the region annotations;
- persist profiler distortion explicitly; profiler timing cannot serve as systems evidence.

For each region invocation and aggregated region family, persist at minimum:
- CPU annotation elapsed time;
- chronological CUDA activity intervals whose timestamps fall within the region annotation;
- unioned CUDA device-active time;
- region-local inter-device idle total/count/median/p95/max;
- region residual `annotation_elapsed - unioned_device_active`, explicitly labeled diagnostic host/control/idle estimate;
- CUDA runtime launch/synchronize/memcpy counts exposed within the annotation;
- invocation count and selected population/batch where available directly from frozen routing metadata;
- whether the region is nested and whether child time has been subtracted.

Avoid double counting: the decision rule must use either leaf regions or parent-exclusive residuals. Persist both raw nested annotations and an explicitly non-overlapping/exclusive attribution table.

Normalize profiler-derived region residuals only by `unprofiled_whole_call_median / profiled_whole_call_cuda_event_latency`, cap any normalized region contribution at the unprofiled whole-call median, and label it diagnostic estimate.

## Preregistered interpretation rule

A future result may name at most one `next_target_region`, and it creates no optimization authority automatically.

A region/family qualifies only if, at BOTH batch 8 and batch 64:
- it is directly observed (not inferred/unavailable);
- exclusive distortion-normalized host/control/idle contribution is at least `1.0 ms`;
- contribution is at least `10%` of that batch's unprofiled whole-call median;
- it is the largest common qualifying exclusive region at both batches.

Allowed target classes:
- `stage_route_control`
- `expert_dispatch_control`
- `latent_reasoner_control`
- `ficem_read_write_control`
- `write_select_update_control`
- `chunk_or_stage_glue_control`
- another exact stable region name only if preregistered in the frozen harness before GPU execution.

If no common region satisfies the rule, return `next_target_region=None` and report batch-specific rankings. Unavailable/unattributed residual cannot authorize an optimization.

## Governance

CPU-first. Before any implementation branch/ref move:
1. fresh-audit live main/issues/PRs;
2. create only issue-specific diagnostic launcher/workflow/CPU-static-test blobs as detached Git objects;
3. build detached tree/commit from exact source main/tree;
4. freeze exact parent #735 lineage, checkpoints, inherited blobs, regions, thresholds, fresh result path/prefixes, new blobs/tree/commit in one comment;
5. only then create the named branch and immutable PR;
6. consume exactly one original full-repository CPU CI attempt; never rerun/retry/redispatch.

After CPU-green merge and fresh exact-main audit: exactly one fresh zero-GPU preauthorization trigger, separate exact-main L4 authorization, then exactly one fresh L4 diagnostic trigger.

No optimization, E2E systems rerun, systems PASS, architecture freeze, S2, fresh scientific seed, independent replication, scaling, or breakthrough claim is authorized by this issue.

**Comment by vinceackermann2-sys** [comment]
## #736 pre-implementation stage/control-flow idle attribution freeze

SOURCE_MAIN=d15d94f39e59d0c6cc8d1514f44a33452d279a74
SOURCE_TREE=fe5181034b3cb07076b17a43c7137829124d8cbb
BRANCH=research/aera-v26-10-issue736-stage-control-idle-attribution
PREAUTH_PREFIX=[aera-v26-10-issue736-region-idle-preauth]
L4_PREFIX=[aera-v26-10-issue736-region-idle-l4]
AUTHORIZATION_HEADING=## #736 sole L4 stage/control-flow idle attribution authorization
RESULT_PATH=/vol/aera-v26/issue736-stage-control-idle-attribution/result.json
PARENT_RESULT_PATH=/vol/aera-v26/issue732-post731-differential-kernel-time-attribution/result.json

PARENT_TRIGGER=735
PARENT_EVIDENCE_COMMENT=5574994790
PARENT_RUN=34156438962
PARENT_JOB=101849137998
PARENT_ATTEMPT=1
PARENT_NEXT_TARGET=None

ISSUE710_LAUNCHER_BLOB=b885250753ea169cd89dd7a978bb3647fd261fe8
V26_10_IMPL_BLOB=d8f691c198eed1fa96bcbb78a4e76cad82d18779
CHECKPOINT_AERA_SHA256=f8aa92421801e8f190247e420632be5f0c20bc5ea8bf6bdeefe06686b3a31b30
CHECKPOINT_TRANSFORMER_SHA256=cdd5cab4439a709468d6607d45d82081b33e876b2e40d91d4a38ba139b219dd7

BATCHES=8,64
WARMUP_CALLS=3
UNPROFILED_CALLS=3
PROFILE_CALLS=1
MIN_REGION_MS=1.0
MIN_REGION_SHARE=0.10
ALLOWED_TARGETS=stage_route_control,expert_dispatch_control,latent_reasoner_control,ficem_read_write_control,write_select_update_control,chunk_or_stage_glue_control,context_attention_control,norm_start_controller_control,end_controller_control,reason_to_chunk_output_norm_control,recurrent_stream_update_control
HISTORICAL_EXCLUSIONS=#405,#258,#731

LAUNCHER_BLOB=2a004bb09eed8f644a61353f762f3ce1bd7d14f2
WORKFLOW_BLOB=beaeaae0558a1208ae96fd555f7900f8e3813ee5
CPU_TEST_BLOB=0c4a903ff57b42ec349080427c65aa5c67fda708
FROZEN_TREE=4c087adac1bbd64f9be0f213da5240b9f276448a
FROZEN_COMMIT=e2766c9f0f818f0526336ca2e9957c842907820e

Frozen diff: exactly 3 additive diagnostic/contract files, 699 additions, 0 deletions. No model/runtime/scientific file is modified.

The diagnostic uses additive `record_function`/MethodType instrumentation only; persisted attribution is an exclusive non-overlapping partition with child intervals subtracted. Each invocation/family records CPU annotation time, device-active union, region-local inter-device idle total/statistics, exclusive residual, runtime counts, and distortion-normalized diagnostic host/control/idle estimate. Unavailable ownership is reported explicitly rather than inferred. The interpretation rule may name at most one common directly observed region meeting >=1.0 ms and >=10% at both batches; it creates no optimization authority automatically.

Superseded detached draft objects `b8b08642686a8127329e875389773b9725447f16` (launcher) and `a11468433153e7a8d75019c40162538135cec707` (workflow) are inert/unreferenced and MUST NOT be used. Earlier unrelated detached #732 objects/commit probes likewise have no #736 authority.

This freeze precedes any #736 branch/ref move. It authorizes creation of only the named branch and one immutable PR for one original full-repository CPU CI attempt. It authorizes no Modal/GPU execution, optimization, E2E systems rerun, systems PASS, architecture freeze, S2, fresh scientific seed, independent replication, scaling, or breakthrough claim.

**Comment by vinceackermann2-sys** [comment]
## #736 sole L4 stage/control-flow idle attribution authorization

Authorize main: `24ae68c393e098ec7535276e58776ea8ef11e238`

Authorize exactly one fresh diagnostic trigger under `[aera-v26-10-issue736-region-idle-l4]` on this exact merged main/tree only.

Prerequisites are satisfied exactly:
- authoritative #736 original freeze comment `5575217545`;
- #738 correction freeze comment `5575502404`;
- merged #739 tree `94b7f64fc4c98f459d6fd45ec2b31f9ac3b9218d`;
- unchanged scientific launcher blob `2a004bb09eed8f644a61353f762f3ce1bd7d14f2`;
- corrected workflow blob `7452335e32c22e778f58519b85ed48e28a1aff36`;
- corrected CPU contract blob `21dce6044ac2097cf5d71d0f66733947d3b7d978`;
- frozen #710 launcher blob `b885250753ea169cd89dd7a978bb3647fd261fe8`;
- frozen v26.10 implementation blob `d8f691c198eed1fa96bcbb78a4e76cad82d18779`;
- sole zero-GPU preauth trigger #740;
- canonical preauth workflow run `34199885289`;
- preauth job `101976023277`;
- attempt `1`;
- preauth evidence comment `5581115526`;
- preauth payload proved `result_absent=true`, `gpu_used=false`, `model_constructed=false`, `new_measurement_performed=false`;
- checkpoint hashes revalidated as AERA `f8aa92421801e8f190247e420632be5f0c20bc5ea8bf6bdeefe06686b3a31b30` and Transformer `cdd5cab4439a709468d6607d45d82081b33e876b2e40d91d4a38ba139b219dd7`;
- all preauth authority flags remained false.

The sole authorized L4 execution is the frozen #736 diagnostic contract only: batches `(8,64)`, 3 warmups, 3 unprofiled whole-call CUDA-event timings, exactly 1 Kineto profiled whole call per batch, hard sparse routing, `update_memory=True`, frozen seed/checkpoints, correctness/route/logit/state/schema/immutability gates, exclusive non-overlapping region attribution, `MIN_REGION_MS=1.0`, `MIN_REGION_SHARE=0.10`, and the preregistered `next_target_region` decision rule.

The result path remains `/vol/aera-v26/issue736-stage-control-idle-attribution/result.json` and must be written at most once. The L4 trigger and its Actions/Modal attempt are consumed regardless of outcome and must never be rerun/retried/redispatched.

This authorization grants no optimization implementation, no full E2E systems rerun, no systems PASS, no architecture freeze, no S2, no fresh scientific seed, no independent replication credit, no scaling, and no breakthrough claim. Even a non-null `next_target_region` requires a separately preregistered optimization successor.
